internal/cmd/server.go

```diff
@@ -84,11 +86,14 @@ func server(ctx context.Context, logOut io.Writer) error {
 		return err
 	}
 
-	// Wait exit signal
+	// Wait for the context to be done, which usually happens the process receives a SIGTERM or SIGINT.
 	<-ctx.Done()
```

Comment:
note on why `<-runnersDone` is needed; Right after ctx is canceled, this main goroutine will call `os.Exit(0)` vs runners are not running on the main goroutine. so without blocking and waiting for it to return, this will highly likely leave some undesirable state left orphaned that is supposed to be cleaned up by non-main goroutine